gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxForm,
+  NxFormGroup,
+  NxToggle,
+  useToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+const DarkModeModal = () => {
+  const [showModal, setShowModal] = useState(false);
+  const modalCloseHandler = () => setShowModal(false);
+
+  const [enableModeChange, toggleEnableModeChange] = useToggle(false);
+  const [mode, setMode] = useState<string | null>(null);
+
+  const { classList } = document.documentElement;
+
+  useEffect(() => {
+    classList.toggle('nx-html--enable-color-schemes', enableModeChange);
+  }, [enableModeChange]);
+
+  useEffect(() => {
+    classList.toggle('nx-html--dark-mode', mode === 'dark' && enableModeChange);
+    classList.toggle('nx-html--light-mode', mode === 'light' && enableModeChange);
+  }, [enableModeChange, mode]);
+
+  return (
+    <>
+      <NxButton variant="tertiary" onClick={() => setShowModal(true)}>Theme Settings</NxButton>
+      { showModal &&
+        <NxModal id="nx-modal-dark-mode-example"
+                 onCancel={modalCloseHandler}
+                 aria-labelledby="nx-modal-dark-mode-form-header">
+          <NxForm onSubmit={modalCloseHandler}
+                  submitBtnText= "Save and Close">
+            <header className="nx-modal-header">
+              <h2 className="nx-h2" id="nx-modal-dark-mode-form-header">
+                Setting Preferences for Light or Dark Mode
+              </h2>
+            </header>
+            <div className="nx-modal-content">
+              <NxFormGroup label="Enable Theme Changes">
+                <NxToggle onChange={toggleEnableModeChange} isChecked={enableModeChange}>
+                  Opt-in to Allow Theming
+                  <NxTooltip title="Opting in will allow you to make adjustments to the display theme. Failure
+                             to do so will always result in light mode. This can be done by adding the
+                             'nx-html--enable-color-schemes' class to the HTML element.">
```

Comment:
I think we could phrase these in a way that makes it more clear that toggling the input will in fact make the change to the classname; the current phrasing almost makes it sound like the user could/should do something _in addition_ to toggling the toggle.  For instance, this one could say "This operates by toggling the presence of the nx-html--enable-color-schemes class"

Reply:
> I think we could phrase these in a way that makes it more clear that toggling the input will in fact make the change to the classname; the current phrasing almost makes it sound like the user could/should do something _in addition_ to toggling the toggle. For instance, this one could say "This operates by toggling the presence of the nx-html--enable-color-schemes class"

Thanks for pointing that out, I hadn't thought about it that way. It has been changed here: [e43333b](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/e43333b0e5ea78c9c661c52d2485325d30fdd5c9)